web-common/src/features/charts/render/VegaEmbed.svelte

```diff
@@ -0,0 +1,147 @@
+<script lang="ts">
+  import type { EmbedOptions, Result } from "vega-embed";
+  import type { SignalListeners, View, VisualizationSpec } from "./types";
+
+  import { createEventDispatcher, onDestroy } from "svelte";
+  import vegaEmbed from "vega-embed";
+
+  import { WIDTH_HEIGHT } from "./constants";
+  import {
+    addSignalListenersToView,
+    combineSpecWithDimension,
+    computeSpecChanges,
+    removeSignalListenersFromView,
+    shallowEqual,
+    updateMultipleDatasetsInView,
+  } from "./utils";
+
+  export let options: EmbedOptions;
+  export let spec: VisualizationSpec;
+  export let view: View | undefined;
+  export let signalListeners: SignalListeners = {};
+  export let data: Record<string, unknown> = {};
+
+  const dispatch = createEventDispatcher();
+
+  let result: Result | undefined = undefined;
+  let prevOptions: EmbedOptions = {};
+  let prevSignalListeners: SignalListeners = {};
+  let prevSpec: VisualizationSpec = {};
+  let prevData: Record<string, unknown> = {};
+  let chartContainer: HTMLElement;
+
+  $: {
+    if (!shallowEqual(data, prevData)) {
+      update();
+    }
+    prevData = data;
+  }
+
+  $: {
```

Comment:
There's a lot going on in this reactive statement. Could you somehow make it more readable, e.g. with comments or smaller functions?